modules/nlu/src/backend/storage.ts

```diff
@@ -185,10 +196,17 @@ export default class Storage {
     return !!_.find(models, m => m.hash === modelHash)
   }
 
-  async getModelAsBuffer(modelHash: string): Promise<Buffer> {
-    const models = await this.ghost.directoryListing(this.modelsDir, '*.bin')
-    const modelFn = _.find(models, m => m.indexOf(modelHash) !== -1)!
-
-    return this.ghost.readFileAsBuffer(this.modelsDir, modelFn)
+  async getModelsFromHash(modelHash: string): Promise<Model[]> {
+    const modelsMeta = await this.getAvailableModels()
+    return Promise.all(
+      modelsMeta
+        .filter(meta => meta.hash === modelHash)
+        .map(async meta => {
+          return {
```

Comment:
very picky-ass comment but `Promise.map` could be used and arrow function instead of return

Reply:
I have abs no problem to use Promise.map but you will agree that the followings are exactly the same. The only difference is one use a callback and the other doesn't ... 

```js
return Promise.map(
      modelsMeta.filter(meta => meta.hash === modelHash),
      async meta => ({
        meta,
        model: await this.ghost.readFileAsBuffer(this.modelsDir, meta.fileName)
      })
    )
```
&

```js
return Promise.all(
  modelsMeta
    .filter(meta => meta.hash === modelHash)
    .map(async meta => ({
        meta,
        model: await this.ghost.readFileAsBuffer(this.modelsDir, meta.fileName)
      })
)
```